dill/_dill.py

```diff
@@ -1360,7 +1360,7 @@ def save_module(pickler, obj):
         if hasattr(obj, "__file__"):
             names = ["base_prefix", "base_exec_prefix", "exec_prefix",
                      "prefix", "real_prefix"]
-            builtin_mod = any(obj.__file__.startswith(os.path.normpath(getattr(sys, name)))
+            builtin_mod = any(obj.__file__.startswith(os.path.realpath(getattr(sys, name)))
```

Comment:
Hi, I just came across this PR because I ran into a similar issue.  I tried to run with master but unfortunately, the problem doesn't go away fully.  I tested a minor adjustment to the code.  Is there any way you would consider putting in another minor fix for this, please?  Use `realpath` on `obj.__file__`, too:

```builtin_mod = any(os.path.realpath(obj.__file__).startswith(os.path.realpath(getattr(sys, name)))```

Reply:
Sure. why don't you submit a PR?